main.go

```diff
@@ -80,14 +80,35 @@ func main() {
 			Usage:       "The token used to push kubeconfigs to GitHub if you need this feature",
 			EnvVar:      "GITHUB_TOKEN",
 			Value:       "",
-			Destination: &cfg.GithubToken,
+			Destination: &cfg.GithubConfig.GithubToken,
 		},
 		cli.StringFlag{
 			Name:        "github-url",
 			Usage:       "The GitHub URL if you are using GitHub enterprise",
 			EnvVar:      "GITHUB_URL",
 			Value:       "",
-			Destination: &cfg.GithubURL,
+			Destination: &cfg.GithubConfig.GithubURL,
+		},
+		cli.StringFlag{
+			Name:        "github-app-private-key-file",
+			Usage:       "GitHub private key file if you are using App based authentication",
+			EnvVar:      "GITHUB_APP_PRIVATE_KEY_FILE",
+			Value:       "",
+			Destination: &cfg.GithubConfig.GithubPrivateKeyFile,
+		},
+		cli.Int64Flag{
+			Name:        "github-app-id",
+			Usage:       "GitHub app id if you are using App based authentication",
+			EnvVar:      "GITHUB_APP_ID",
+			Value:       0,
+			Destination: &cfg.GithubConfig.GithubAppID,
+		},
+		cli.Int64Flag{
+			Name:        "github-app-installation-id",
+			Usage:       "GitHub app installation id if you are using App based authentication",
+			EnvVar:      "GITHUB_APP_INSTALLATION_ID",
+			Value:       0,
+			Destination: &cfg.GithubConfig.GithubInstallationID,
```

Comment:
The installation ID isn't really something the user is expected to supply. It can be different for different repos, and are for sure always different for different orgs. Suggest evaluating the installation ID dynamically per upload instead of a flag.

As described in the docs, you can get the installation ID from APIs:

- `GET /users/{username}/installation` get installation for user
- `GET /orgs/{org}/installation` get installation for org
- `GET /repos/{owner}/{repo}/installation` get installation for specific repo
- `GET /app/installations` to list all installations

Some docs:

- <https://docs.github.com/en/enterprise-server@3.9/apps/creating-github-apps/authenticating-with-a-github-app/generating-an-installation-access-token-for-a-github-app#generating-an-installation-access-token>
- <https://docs.github.com/en/enterprise-server@3.9/rest/apps/apps?apiVersion=2022-11-28#get-a-repository-installation-for-the-authenticated-app>

Reply:
Fixed in [9ece835](https://github.com/jadolg/klum/pull/25/commits/9ece83560e062c5304c097ceb65fb0ade91bfc19)